tools/rosbag_storage/CMakeLists.txt

```diff
@@ -24,21 +24,40 @@ add_definitions(-D_FILE_OFFSET_BITS=64)
 include_directories(include ${catkin_INCLUDE_DIRS} ${console_bridge_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS} ${BZIP2_INCLUDE_DIR})
 add_definitions(${BZIP2_DEFINITIONS})
 
-add_library(rosbag_storage
-  src/bag.cpp
-  src/bag_player.cpp
-  src/buffer.cpp
-  src/bz2_stream.cpp
-  src/lz4_stream.cpp
-  src/chunked_file.cpp
-  src/message_instance.cpp
-  src/query.cpp
-  src/stream.cpp
-  src/view.cpp
-  src/uncompressed_stream.cpp
-)
-
-target_link_libraries(rosbag_storage ${catkin_LIBRARIES} ${Boost_LIBRARIES} ${BZIP2_LIBRARIES} ${console_bridge_LIBRARIES})
+if(WIN32)
+  add_library(rosbag_storage
+    src/bag.cpp
+    src/bag_player.cpp
+    src/buffer.cpp
+    src/bz2_stream.cpp
+    src/lz4_stream.cpp
+    src/chunked_file.cpp
+    src/encryptor.cpp
+    src/message_instance.cpp
+    src/query.cpp
+    src/stream.cpp
+    src/view.cpp
+    src/uncompressed_stream.cpp
+  )
+  target_link_libraries(rosbag_storage ${catkin_LIBRARIES} ${Boost_LIBRARIES} ${BZIP2_LIBRARIES} ${console_bridge_LIBRARIES})
+else()
+  add_library(rosbag_storage
+    src/aes_encryptor.cpp
+    src/bag.cpp
+    src/bag_player.cpp
+    src/buffer.cpp
+    src/bz2_stream.cpp
+    src/lz4_stream.cpp
+    src/chunked_file.cpp
+    src/encryptor.cpp
+    src/message_instance.cpp
+    src/query.cpp
+    src/stream.cpp
+    src/view.cpp
+    src/uncompressed_stream.cpp
+  )
+  target_link_libraries(rosbag_storage ${catkin_LIBRARIES} ${Boost_LIBRARIES} ${BZIP2_LIBRARIES} ${console_bridge_LIBRARIES} crypto gpgme)
+endif()
```

Comment:
Please do not duplicate large chunks of code like this. Instead create a variable for the optional files, fill that variable based on the platform, and then use it as one argument to the `add_library` call.

Reply:
Addressed.  Thank you.